source/magpie/index.d

```diff
@@ -364,12 +370,14 @@ public:
 
```

Comment:
(sorry I can't seem to comment higher up)

This loop should be using `foreach(i, ref e; indexing[axis].codes)`

`e` is then a reference to `indexing[axis].codes[i]`. Possibly consider also consider iterating `indexing[axis].index` and `indexing[axis].codes` in [lockstep](https://dlang.org/phobos/std_range.html#lockstep)

Reply:
As of now I have used ref. I'll look into lockstep and implement that in the next refactoring stage.